account_page.php

```diff
@@ -181,13 +181,16 @@
 			if( $t_account_verification ) {
 				token_set( TOKEN_ACCOUNT_VERIFY, true, TOKEN_EXPIRY_AUTHENTICATED, $u_id );
 			} else {
+				# If the current password is blank, do not require the field
+				# so the user can reset the password (#23507)
+				$t_required = auth_does_password_match( $u_id, '' ) ? '' : 'required';
 			?>
 			<tr>
 				<td class="category">
-					<span class="required"><?php if( $t_force_pw_reset ) { ?> * <?php } ?></span> <?php echo lang_get( 'current_password' ) ?>
+					<span class="<?php echo $t_required ?>"><?php if( $t_force_pw_reset ) { ?> * <?php } ?></span> <?php echo lang_get( 'current_password' ) ?>
```

Comment:
I would rather not generate `class=""` when field is not required.  Not sure if this would upset some validator.  Maybe also create `$t_required_class` and use it here.

Reply:
`class=""` is [valid HTML](https://www.w3.org/TR/html5/dom.html#classes) so I really don't see the point of adding code complexity, or overhead of an additional variable.